ocaml/xe-cli/newcli.ml

```diff
@@ -349,11 +349,11 @@ let main_loop ifd ofd =
           marshal ofd (Response OK);
           let length = (Unix.stat x).Unix.st_size in
           marshal ofd (Blob (Chunk (Int32.of_int length)));
-          let buffer = String.make (1024 * 1024 * 10) '\000' in
+          let buffer = Bytes.make (1024 * 1024 * 10) '\000' in
           let left = ref length in
           while !left > 0 do
-            let n = Unix.read fd buffer 0 (min (String.length buffer) !left) in
-            Xapi_stdext_unix.Unixext.really_write ofd buffer 0 n;
+            let n = Unix.read fd buffer 0 (min (Bytes.length buffer) !left) in
+            Xapi_stdext_unix.Unixext.really_write ofd (Bytes.unsafe_to_string buffer) 0 n;
```

Comment:
is there a `really_write_substring`

Reply:
`really_write_string`, I had forgotten about it

Reply:
lets keep it as it is for, we can update unixext if we see patterns like this a lot in the future.